### PR TITLE
fix credential bugs

### PR DESCRIPTION
**Pull request by yiliuTo**
This pr is to fix credential related bugs including:
1. Fail to use resource manager to retrieve connection string.
2. Default token credential still takes effects when other token credentials are provided.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java [hunk 1]
```diff
@@ -40,21 +43,31 @@
...
         NamespaceProperties namespaceProperties = new NamespaceProperties();
-        AzurePropertiesUtils.copyAzureCommonProperties(properties, namespaceProperties);
         BeanUtils.copyProperties(properties, namespaceProperties);
         copyAzureCommonProperties(properties, namespaceProperties);
+        if (namespaceProperties.getConnectionString() == null) {
+            ServiceConnectionStringProvider<AzureServiceType.EventHubs> connectionStringProvider =
+                connectionStringProviders.getIfAvailable();
+            if (connectionStringProvider != null) {
+                namespaceProperties.setConnectionString(connectionStringProvider.getConnectionString());
+                LOGGER.warn("Event Hubs connection string is set now.");
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java:68]: change to debug / info,` "Event Hubs connection string is set from {} now.", connectionStringProvider.getClass().getName()`

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusMessagingAutoConfiguration.java [hunk 2]
```diff
@@ -45,12 +49,22 @@
...
+                                                      ObjectProvider<ServiceConnectionStringProvider<AzureServiceType.ServiceBus>> connectionStringProviders) {
         NamespaceProperties namespaceProperties = new NamespaceProperties();
         BeanUtils.copyProperties(properties, namespaceProperties);
         copyAzureCommonProperties(properties, namespaceProperties);
+        if (namespaceProperties.getConnectionString() == null) {
+            ServiceConnectionStringProvider<AzureServiceType.ServiceBus> connectionStringProvider =
+                connectionStringProviders.getIfAvailable();
+            if (connectionStringProvider != null) {
+                namespaceProperties.setConnectionString(connectionStringProvider.getConnectionString());
+                LOGGER.warn("Service Bus connection string is set now.");
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusMessagingAutoConfiguration.java:65]: same here

File: sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/ServiceBusMessageChannelBinder.java [hunk 3]
```diff
@@ -134,7 +134,7 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
         ServiceBusContainerProperties containerProperties = new ServiceBusContainerProperties();
         containerProperties.setEntityName(destination.getName());
         containerProperties.setSubscriptionName(group);
-
+        containerProperties.setAutoComplete(properties.getExtension().getAutoComplete());
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/ServiceBusMessageChannelBinder.java:137]: we should use another PR to fix this.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java (hunk 1 above)
Comment (yiliuTo) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java:68]: Cool, good idea, we can avoid print the connection string and also show users more info in this way.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**